contracts/interfaces/draft-IERC7786.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+/**
+ * @dev Interface for ERC-7786 source gateways.
+ *
+ * See ERC-7786 for more details
+ */
+interface IERC7786GatewaySource {
+    /**
+     * @dev Event emitted when a message is created. If `outboxId` is zero, no further processing is necessary, and
+     * no {MessageSent} event SHOULD be expected. If `outboxId` is not zero, then further (gateway specific, and non
+     * standardized) action is required.
+     */
+    event MessageCreated(
+        bytes32 outboxId,
```

Comment:
I think this should be `indexed`. I know it's not in the spec but we should fix that.

Reply:
I think we discussed it. 

The MessageSent one is indexed (because we may want to activelly filter a particular MessageSent once we know the id. But for the `MessageCreated` part, we don't have a way to predict the value (at least not in general) so creating a is not that meaningfull IMO (hence why its not indexed)

Reply:
My reasoning was that if you see a `MessageSent` event you may want to retrieve the matching `MessageCreated`.

Reply:
make sens